### PR TITLE
add gke_backup_agent_config to google_container_cluster

### PR DESCRIPTION
**Pull request by miticojo**
This PR fixes [10970](https://github.com/hashicorp/terraform-provider-google/issues/10970) and add support for [GKE Backup](https://cloud.google.com/kubernetes-engine/docs/add-on/backup-for-gke/concepts/backup-for-gke) Agent addon in google_container_cluster.

Usage:
```
resource "google_container_cluster" "gke" {
  provider = google-beta
  ...
  addons_config {
     gke_backup_agent_config {
      enabled = true
    }
  }
}
```
If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
container: added `gke_backup_agent_config` in `addons_config` to `google_container_cluster` (beta)
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
Is this feature available? I don't see it within the public documentation for the rest api
https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#Cluster.AddonsConfig

**Comment by miticojo** [comment]
> Is this feature available? I don't see it within the public documentation for the rest api https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1/projects.locations.clusters#Cluster.AddonsConfig

It's available in the v1beta1
https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.locations.clusters#Cluster.AddonsConfig

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
Could you also add documentation for this field to 
https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/website/docs/r/container_cluster.html.markdown

File: mmv1/third_party/terraform/resources/resource_container_cluster.go.erb [hunk 1]
```diff
@@ -71,6 +71,7 @@ var (
 		"addons_config.0.gce_persistent_disk_csi_driver_config",
 		"addons_config.0.kalm_config",
 		"addons_config.0.config_connector_config",
+        "addons_config.0.gke_backup_agent_config",
```
[the remark this answers is not in the document]
Comment (miticojo) [reply mmv1/third_party/terraform/resources/resource_container_cluster.go.erb:74]: Sure, thanks

**Review by ScottSuarez — APPROVED** [review]
lgtm

**Comment by ScottSuarez** [comment]
Could you update the release note to follow the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md?rgh-link-date=2022-04-13T17%3A04%3A16Z)?

**Comment by miticojo** [comment]
> Could you update the release note to follow the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md?rgh-link-date=2022-04-13T17%3A04%3A16Z)?

Done!

**Comment by miticojo** [comment]
Sorry I've closed the PR by mistake

**Pull request merged**